pennylane/ops/qubit/parametric_ops.py

```diff
@@ -1269,6 +1269,182 @@ def control_wires(self):
         return self.wires[0:1]
 
 
+class PCPhase(Operation):
+    r"""
+    A Projector Controlled Phase gate.
+
+    .. math:: \Pi_{\phi}(\phi, dim=2, wires=[0,1]) = \begin{bmatrix}
+                e^{i\phi} & 0 & 0 & 0 \\
+                0 & e^{i\phi} & 0 & 0 \\
+                0 & 0 & 1 & 0 \\
+                0 & 0 & 0 & 1
+            \end{bmatrix}.
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        dim (int): the qubit subspace on which we apply the phase shift
+        wires (Any, Wires): the wire the operation acts on
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+
+    **Details:**
+
+    * Number of wires: AnyWires
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+    * Gradient recipe: (~ ADD THIS ~)
```

Comment:
We add the Gradient recipe in this PR?

Reply:
After an initial conversation with @dwierichs, we concluded that we don't need to include it. There will be a 2nd PR to add the adjoint gradient method, we can include it in there.